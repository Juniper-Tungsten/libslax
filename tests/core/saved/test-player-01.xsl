<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="1.0">
  <xsl:output indent="yes"/>
  <xsl:template match="/roster">
    <out>
      <xsl:variable name="x">
        <xsl:apply-templates/>
      </xsl:variable>
      <x>
        <xsl:copy-of select="$x"/>
      </x>
    </out>
  </xsl:template>
  <xsl:template match="player">
    <xsl:if test="salary">
      <player last="{last}" first="{first}">
        <salary>
          <xsl:value-of select="salary"/>
        </salary>
        <xsl:variable name="salary" select="salary"/>
        <count>
          <xsl:value-of select="count(../player[salary &gt; $salary])"/>
        </count>
      </player>
    </xsl:if>
  </xsl:template>
  <xsl:template match="text()"/>
</xsl:stylesheet>
