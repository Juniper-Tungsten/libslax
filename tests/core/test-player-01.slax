version 1.0;

output-method {
    indent "yes";
}

match /roster {
    <out> {
	var $x = {
	    apply-templates;
	}
        <x> { copy-of $x; }
    }
}

match player {
    if (salary) {
	<player last=last first=first> {
	    <salary> salary;
	    var $salary = salary;
	    <count> count(../player[salary > $salary]);
	}
    }
}

match text() { }
