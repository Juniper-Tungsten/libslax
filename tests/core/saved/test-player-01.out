<?xml version="1.0"?>
<out>
  <x>
    <player last="Cole" first="Erik">
      <salary>2800000</salary>
      <count>4</count>
    </player>
    <player last="Corvo" first="Joe">
      <salary>2750000</salary>
      <count>5</count>
    </player>
    <player last="Gleason" first="Tim">
      <salary>2500000</salary>
      <count>6</count>
    </player>
    <player last="Harrison" first="Jay">
      <salary>500000</salary>
      <count>11</count>
    </player>
    <player last="Jokinen" first="Jussi">
      <salary>1500000</salary>
      <count>8</count>
    </player>
    <player last="Kostopoulos" first="Tom">
      <salary>700000</salary>
      <count>10</count>
    </player>
    <player last="LaRose" first="Chad">
      <salary>1500000</salary>
      <count>8</count>
    </player>
    <player last="O'Sullivan" first="Patrick">
      <salary>2387500</salary>
      <count>7</count>
    </player>
    <player last="Pitkanen" first="Joni">
      <salary>4000000</salary>
      <count>1</count>
    </player>
    <player last="Ruutu" first="Tuomo">
      <salary>3000000</salary>
      <count>3</count>
    </player>
    <player last="Staal" first="Eric">
      <salary>6000000</salary>
      <count>0</count>
    </player>
    <player last="Ward" first="Cam">
      <salary>3500000</salary>
      <count>2</count>
    </player>
  </x>
</out>
